src/wallet/wallet.cpp

```diff
@@ -4374,6 +4374,15 @@ bool CWallet::SetActiveHDKey(const CExtPubKey& xpub)
     return WalletBatch(GetDatabase()).WriteActiveHDKey(xpub);
 }
 
+std::optional<CExtPubKey> CWallet::GetActiveHDPubKey()
+{
+    AssertLockHeld(cs_wallet);
+    if (m_xpub.pubkey.IsValid()) {
```

Comment:
in e693bd02610df20f73c008b03c739da5d69fcf76

nit: If the key is not valid - it should be an error and it should be detected in `LoadActiveHDKey()`

Reply:
Added `Assume`